geonode_mapstore_client/client/js/components/DetailsPanel/DetailsLocations.jsx

```diff
@@ -0,0 +1,179 @@
+
+    const polygon = !isEmpty(extent) ? getPolygonFromExtent(extent) : null;
+    const center = !isEmpty(extent) && polygon ? turfCenter(polygon) : null;
+    const isDrawn = !isEqual(initialExtent, extent);
+
+    return (
+        <div className="gn-viewer-extent-map">
+            <BoundingBoxAndCenter center={center} extent={extent} expand={!allowEdit}/>
+            {allowEdit && <Map
+                id="gn-locations-map"
+                mapType={"openlayers"}
+                map={{
+                    registerHooks: false,
+                    projection: "EPSG:4326"
+                }}
+                styleMap={{
+                    width: "100%",
+                    height: "100%"
+                }}
+                layers={[
+                    {
+                        type: 'osm',
+                        title: 'Open Street Map',
+                        name: 'mapnik',
+                        source: 'osm',
+                        group: 'background',
+                        visibility: true
+                    },
+                    ...(!isEmpty(extent)
+                        ? [
+                            {
+                                id: "extent-location",
+                                type: "vector",
+                                features: [
+                                    {
+                                        ...polygon,
+                                        style: {
+                                            color: isDrawn ? "#ffaa01" : "#397AAB",
+                                            opacity: 0.8,
+                                            fillColor: isDrawn
+                                                ? "rgba(255, 170, 1, 0.1)"
+                                                : "#397AAB",
+                                            fillOpacity: 0.2,
+                                            weight: 2
+                                        }
+                                    },
+                                    {
+                                        ...center,
+                                        style: {
+                                            iconGlyph: "sort-desc",
+                                            iconShape: "circle",
+                                            iconColor: isDrawn ? "yellow" : "blue"
+                                        }
+                                    }
+                                ]
+                            }
+                        ]
+                        : [])
+                ]}
+            >
+                <ZoomTo extent={extent?.join(",")} />
+                <DrawSupport onSetExtent={onSetExtent}/>
+            </Map>
+            }
+        </div>
+    );
+};
+
+const ConnectedDetailsLocations = connect(
+    null,
+    {
+        onSetExtent: setResourceExtent
+    }
+)(DetailsLocations);
+
+export default ConnectedDetailsLocations;
```

Comment:
Components should not be connected, connections should be in plugins, pages or containers

Reply:
My bad, never meant to have it. Missed to refactor at commit.